util/fix_include_guard.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# fix_include_guard.py script takes any number of C or C++ header files as
+# input, and formats their include guards to conform to the style mandated
+# by Google C++. See
+# https://google.github.io/styleguide/cppguide.html#The__define_Guard
+#
+# clang-format does not, unfortunately, have support for automatically
+# generating the correct include guards; hence the existence of this script.
+#
+# This script will write the names of all affected files to stdout; if no such
+# output is present, all files have the correct guards. This script is
+# idempotent. If it cannot fix a file, the script will write the file name to
+# stderr and will exit at the end with a nonzero status code.
+
+import argparse
+import re
+import sys
+from pathlib import Path
+
+PROJECT_NAME = "OPENTITAN_PROVISIONING"
+
+# This file is $REPO_TOP/util/fix_include_guard.py, so it takes two parent()
+# calls to get back to the top.
+REPO_TOP = Path(__file__).resolve().parent.parent
```

Comment:
Other generalization TODOs:
Accept the project name as a cmdline parameter; plumb through to the rule attrs.
Accept the repo top as a cmdline parameter; use the `workspace` attr to learn the location of the repo top.

Reply:
Done. Tracked in #41.